### PR TITLE
Move vault code to hashicorp vault plugin, move tests to integrations

### LINKED ISSUE #968: Move Vault support to another plugin
There is a https://github.com/jenkinsci/hashicorp-vault-plugin plugin which implements Vault functionality for Jenkins. It uses Vault API libraries as well as the JCasC plugin, and there are binary compatibility issues: https://github.com/jenkinsci/configuration-as-code-plugin/issues/922

There is no practical reason to depend on a specific technology provider library in a gneric JCasC plugin. As suggested in https://github.com/jenkinsci/hashicorp-vault-plugin/pull/23 , I suggest moving the code to a separate plugin or to Hashicorp Vault Plugin directly.

It could also help us to ensure that Secret Source extensions are really extensible

### PR DESCRIPTION
**Pull request by casz**
### Your checklist for this pull request

🚨 Please review the [guidelines for contributing](../blob/master/docs/CONTRIBUTING.md) to this repository.

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side) and not your master branch!

- [x] Please describe what you did

- [x] Link to relevant GitHub issues or pull requests

- [x] Link to relevant [Jenkins JIRA issues](https://issues.jenkins-ci.org)

- [x] Did you provide a test-case? That demonstrates feature works or fixes the issue.

### Description

Moved the vault secret handling to https://github.com/jenkinsci/hashicorp-vault-plugin/pull/35 as discussed in #968 

resolves #968

### CODE REVIEW

**Comment by casz** [comment]
@fishi0x01 Hopefully you don't that we are moving the code? :sweat_smile:

**Comment by jtnord** [comment]
What is the driver for this?  last time I looked at the vault plugin it handles credentials in a very non standard way and as such I would not actually like that plugin installed in order to bootstrap credentials in Jenkins with CasC.

**Comment by casz** [comment]
The main driver is to avoid class loading issues between the plugins and it seems like the better home for this code.

**Comment by jtnord** [comment]
> avoid class loading issues between the plugins

either use a speparate api plugin or use plugin-first-classloader?

**Comment by casz** [comment]
`plugin-first-classloader` no thanks
and `hashicorp-vault-plugin` and `jcasc` are they only two depending on `vault-java-driver`

**Comment by casz** [comment]
This way we remove the extra dependency inside `jcasc`

**Comment by jtnord** [comment]
Can you at least make it optional then so that if you don;t want to bootsrap with vault you are not forced to install vault?

**Comment by casz** [comment]
JCasC does not depend on hashicorp-vault-plugin...
Our integration tests do.

JCasC is even optional inside `hashicorp-vault-plugin`
https://github.com/jenkinsci/hashicorp-vault-plugin/pull/35/files#diff-7901bb0f514a3d68fe4cdeb611d100a3R24
https://github.com/jenkinsci/hashicorp-vault-plugin/pull/35/files#diff-600376dffeb79835ede4a0b285078036R178

**Comment by oleg-nenashev** [comment]
Do we want to make 1.26 as incompatible after this change?

**Comment by casz** [comment]
Yes @oleg-nenashev but I would like to implement support for vault agent #967 before releasing.
I think also it would be best to release in tandem. First, release JCasC and then hashicorp-vault-plugin so it depends on the release that has Vault secret support removed.

**Comment by casz** [comment]
I'd also like to test the scenario where someone has forgotten to install hashicorp vault plugin and perhaps alert user they need to install that plugin 😅

**Comment by casz** [comment]
Please take another look @timja or @oleg-nenashev 😅

Perhaps it would be better not to throw an exception or WDYT?

**Comment by casz** [comment]
Well my other tests say no 😆

File: README.md [hunk 1]
```diff
@@ -336,7 +336,7 @@ secrets:
 **TODO**: Provide a Dockerfile to generate documentation from specified jenkins-core release and plugins.
 
 ## Installing plugins
-
+
```
Comment (casz) [inline README.md]:
```suggestion
```

**Comment by casz** [comment]
Will release this first and than release vault.
So adding a follow up PR that switches to release. Again intergration module is never released so doesn't matter if we use incremental versions

**Comment by casz** [comment]
working on fixing the test :)

**Pull request merged**